### PR TITLE
LPS-125276 Use reference as key, as id does not always exist

### PR DESCRIPTION
**Pull request by sandrodw3**

### CODE REVIEW

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
ci:test:sf

**Comment by jeyvison** [comment]
ci:test:relevant

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js [hunk 1]
```diff
@@ -198,35 +198,24 @@ const Options = ({
...
-
-				availableLanguageIds.forEach((languageId) => {
-					normalizedValue[languageId] = value[languageId].map(
-						(option) => {
-							if (option.edited) {
-								return option;
-							}
+		availableLanguageIds.forEach((languageId) => {
+			normalizedValue[languageId] = value[languageId].map((option) => {
+				if (option.edited) {
```
Comment (marcosapmf) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:205]: Hey @sandrodw3, we have a fake option that is show when we don't have any options created yet. In this case, the option value is empty. Did you consider that when you thought on this solution?

Comment (sandrodw3) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:205]: Hi @marcosapmf 🙂 thanks for the feedback. I actually considered that. I saw you have always a default option when there is no other options. It's still there. The only difference with my solution is that you will be able to delete it, just as with the other options, but you will always have at least one option. I think it's consistent.

Anyway, I would be grateful if some of you guays can give it a try and tell me if that behavior is ok for you, or you need any change.

Thanks again! 🙂

Comment (marcosapmf) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Options/Options.es.js:205]: Hey @sandrodw3, thanks for the clarification. 

We'll do that!

Thanks to you

**Comment by ealonso** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
Just started reviewing :) <br><br>:octocat: *Sent from [GH](http://nodegh.io).*

**Comment by marcelabc** [comment]
Hi @sandrodw3 !

We reviewed your PR and we found out:

- There is one frontend test failing:
Screen Shot 2021-01-11 at 12 01 08

- Your solution does not accord to the expected option field behavior. The default option should only appear in the default language. If the user changes the language, the "default option" does not appear. Please, see the video:

https://user-images.githubusercontent.com/15161924/104199502-4a6b1200-5406-11eb-8414-b77447284265.mp4

If you have any questions, please let me know :) Thank you!

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi @sandrodw3 !

We reviewed your PR and we found out:

- There is one frontend test failing:
Screen Shot 2021-01-11 at 12 01 08

- Your solution does not accord to the expected option field behavior. The default option should only appear in the default language. If the user changes the language, the "default option" does not appear. Please, see the video:

https://user-images.githubusercontent.com/15161924/104199502-4a6b1200-5406-11eb-8414-b77447284265.mp4

If you have any questions, please let me know :) Thank you!

**Comment by sandrodw3** [comment]
Hi @marcelabc! 🙂 Thanks for your review! I missed that detail, I'll rethink the solution today.

**Comment by sandrodw3** [comment]
Hi @marcelabc 🙂 

I've been taking a look on this and I have a few doubts. For me, with today's master, it's not working like in your video. In my case:

- If I add the Single Selection field in the default language, it adds the default option to all languages.
- If I add the Single Selection field in a language that is not the default one, it adds the default option to all languages except that one.

https://user-images.githubusercontent.com/2444936/104335400-4238d280-54eb-11eb-8857-33d7493baed8.mov

For me it's a little bit weird, specially the second case. I can try to make it work like you said, but I think it's actually not working that way right now in master. Could you confirm that? Thanks again! 😄

**Comment by sandrodw3** [comment]
ci:test:sf

**Comment by sandrodw3** [comment]
ci:test:relevant

**Comment by sandrodw3** [comment]
Hi @marcelabc 🙂

On the first place, I found out why that test was failing and I've updated my PR, so all frontend tests are passing with success right now.

Secondly, regarding your comment and your video, I have some doubts. For me, with today's master (without my changes), it's not working like you described. In my case:

* If I add the Single Selection field in the default language, it adds the default option to all languages.
* If I add the Single Selection field in a language that is not the default one, it adds the default option to all languages except that one.

https://user-images.githubusercontent.com/2444936/104339386-6eeee900-54ef-11eb-951a-bdf26496757e.mov

For me, it's a little bit weird, specially the second case. With my changes, the default option is being added always, in all languages, that I think is more consistent.

In my opinion, if you prefer the default option to be added only in the default language, we should fill another ticket, as it's actually not working that way on master right now.

What my changes are doing is allowing the user to remove the default option, and I think it solves the bug described on LPS-125276.

What do you think? 😄  Please let me know if tou have a different thought. Thanks a lot again!!

**Review by marcelabc — APPROVED** [review]
I tested now and let the default option appear in all languages did not show any irregular behavior at first. So let's set this behavior until we find any critical bug :) 

LGTM!

**Comment by marcelabc** [comment]
ci:forward

**Pull request closed without merging**